server/aesdsocket.c

```diff
@@ -0,0 +1,204 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <string.h>
+#include <syslog.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#define LOG_FILE_LOC "/var/tmp/aesdsocketdata"
+#define PORT "9000"
+#define MAX_BUFFER_SIZE 256
+
+int sockfd = -1;
+
+void cleanup_and_exit(int status)
+{
+    syslog(LOG_INFO, "Closing aesdsocket application");
+    closelog();
+    close(sockfd);
+    remove(LOG_FILE_LOC);
+    exit(status);
+}
+
+void signal_handler(int signo)
+{
+    syslog(LOG_USER, "Caught signal, exiting");
+    cleanup_and_exit(EXIT_SUCCESS);
+}
+
+void setup_signal_handler()
+{
+    struct sigaction sa;
+    memset(&sa, 0, sizeof(struct sigaction));
+    sa.sa_handler = signal_handler;
+
+    if (sigaction(SIGTERM, &sa, NULL) != 0 || sigaction(SIGINT, &sa, NULL) != 0)
+    {
+        syslog(LOG_ERR, "Error setting up signal handler");
+        cleanup_and_exit(EXIT_FAILURE);
+    }
+}
+
+void handle_client_connection(int client_fd)
+{
+    FILE *fp = fopen(LOG_FILE_LOC, "a+");
+    char buffer[MAX_BUFFER_SIZE];
+
+    while (1)
+    {
+        ssize_t bytes_recv = recv(client_fd, buffer, sizeof(buffer), 0);
+        if (bytes_recv <= 0)
+       	{
+            break;
+        }
+
+        fwrite(buffer, bytes_recv, 1, fp);
```

Comment:
@akka2103 with this design you will write to the file before the packet completes, which may cause problems with future assignments.

Reply:
Does this mean, that I have to first receive till the packet completes. Then write into the file?

Reply:
You can check for a '\n' in each iteration of 256 bytes recv request and determine whether to write entire string to the file or not.